<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Seyoon Ko - Publications</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../teaching/index.html" rel="next">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../publications/index.html">Publications</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-center sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Seyoon Ko</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../publications/index.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Publications</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../teaching/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Teaching</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../software/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Software</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../about.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../cv.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CV</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#peer-reviewed-papers" id="toc-peer-reviewed-papers" class="nav-link active" data-scroll-target="#peer-reviewed-papers">Peer-reviewed Papers</a></li>
  <li><a href="#working-papers-and-preprints" id="toc-working-papers-and-preprints" class="nav-link" data-scroll-target="#working-papers-and-preprints">Working Papers and Preprints</a></li>
  <li><a href="#conference-abstracts" id="toc-conference-abstracts" class="nav-link" data-scroll-target="#conference-abstracts">Conference Abstracts</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Publications</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>Author profile on Google Scholar: <a href="https://scholar.google.com/citations?user=eAPnyCsAAAAJ&amp;hl">link</a></p>
<section id="peer-reviewed-papers" class="level2">
<h2 class="anchored" data-anchor-id="peer-reviewed-papers">Peer-reviewed Papers</h2>
<ul>
<li>Chu, B. B., <strong>Ko, S.</strong>, Zhou, J. J., Jensen, A., Zhou, H., Sinsheimer, J. S., Lange, K. (2023). Multivariate Genomewide Association Analysis by Iterative Hard Thresholding. <em>Bioinformatics</em>, 39(4), btad193. [<a href="https://doi.org/10.1093/bioinformatics/btad193">paper</a>] [<a href="https://github.com/OpenMendel/MendelIHT.jl">Julia code</a>]</li>
<li><strong>Ko, S.</strong>, Chu, B. B., Peterson, D., Okenwa, C., Papp, J. C., Alexander, D. H., Sobel, E. M., Zhou, H., Lange, K. (2023). Unsupervised discovery of ancestry informative markers and genetic admixture proportions in biobank-scale data sets. <em>American Journal of Human Genetics</em>, 109(3), pp.&nbsp;433-445. [<a href="https://doi.org/10.1016/j.ajhg.2022.12.008">paper</a>][<a href="https://github.com/OpenMendel/OpenADMIXTURE.jl">Julia code</a>]</li>
<li><strong>Ko, S.</strong>, Zhou, H., Zhou, J., and Won, J.-H. (2022). High-Performance Statistical Computing in the Computing Environments of the 2020s. <em>Statistical Science</em>, 37(4), pp.&nbsp;494–518. [<a href="https://projecteuclid.org/journals/statistical-science/volume-37/issue-4/High-Performance-Statistical-Computing-in-the-Computing-Environments-of-the/10.1214/21-STS835.short">paper</a>] [<a href="https://github.com/kose-y/dist_stat">Python code, PyTorch, dask</a>]</li>
<li>Kim J., Jensen, A., <strong>Ko, S.</strong>, Raghavan, S., Phillips, L. S., Hung, A., Sun, Y., Zhou, H., Reaven, P., Zhou, J. J. (2022), Systematic Heritability and Heritability Enrichment Analysis for Diabetes Complications in UK Biobank and ACCORD Studies, <em>Diabetes</em>, 71(5), pp.&nbsp;1137–1148.</li>
<li>∗<strong>Ko, S.</strong>, ∗German, C., Jensen, A., Shen, J., Wang, A., Mehrotra, D. V., Sun, Y. V., Sinsheimer, J. S., Zhou, H., Zhou, J. J. (2022), GWAS of longitudinal trajectories at biobank scale, <em>American Journal of Human Genetics</em>, 109(3), pp.&nbsp;433–445. [<a href="https://github.com/OpenMendel/TrajGWAS.jl">Julia code</a>]</li>
<li>Chu, B. B., Sobel, E. M., Wasiolek, R., <strong>Ko, S.</strong>, Sinsheimer, J. S., Zhou, H., Lange, K. (2021), A Fast Data-Driven Method for Genotype Imputation, Phasing, and Local Ancestry Inference: MendelImpute.jl. <em>Bioinformatics</em>, 37(24), pp.&nbsp;4756–4763.</li>
<li><strong>Ko, S.</strong>, Li, G. X., Choi, H. and Won, J.-H. (2021). Computationally scalable regression modeling for ultrahigh-dimensional omics data with ParProx. <em>Briefings in Bioinformatics</em>, 22(6), bbab256. [<a href="https://github.com/kose-y/ParProx.jl">Julia code</a>]</li>
<li>Kim, J., Shen, J., Wang, A., Mehrotra, D. V., <strong>Ko, S.</strong>, Zhou, J. J., Zhou, H. (2021), VCSEL: Prioritizing SNP-Set by Penalized Variance Component Selection, <em>Annals of Applied Statistics</em>, 15(4), pp.&nbsp;1652–1672.</li>
<li>Ryu, E. K., <strong>Ko, S.</strong>, Won, J.-H. (2020). Splitting with Near-Circulant Linear Systems: Applications to Total Variation CT and PET. <em>SIAM Journal on Scientific Computing</em>, 42(1), pp.&nbsp;B185–B206. [<a href="https://github.com/kose-y/near-circulant-splitting">Matlab code</a>]</li>
<li>Zhou, H., Sinsheimer, J. S., Bates, D. M., Chu, B. B., German, C. A., Ji, S. S., Keys, K. L., Kim, J., <strong>Ko, S.</strong>, Mosher, G. D., and Papp, J. C. (2020). OpenMendel: A cooperative programming project for statistical genetics. <em>Human Genetics</em>, 139(1), pp.&nbsp;61–71. [<a href="https://openmendel.github.io/">project homepage</a>]</li>
<li><strong>Ko, S.</strong>, Yu, D., Won, J.-H. (2019). Easily parallelizable and distributable class of algorithms for structured sparsity, with optimal acceleration. <em>Journal of Computational and Grahpical Statistics</em>, 28(4), pp.&nbsp;821–833. [<a href="https://github.com/kose-y/dist-primal-dual">Python code, TensorFlow, Docker</a>]</li>
<li><strong>Ko, S.</strong> and Won, J. H. (2019). Optimal Minimization of the Sum of Three Convex Functions with a Linear Operator. In <em>The 22nd International Conference on Artificial Intelligence and Statistics</em>, pp.&nbsp;1185–1194.</li>
<li>Racah, E., <strong>Ko, S.</strong>, Sadowski, P., Bhimji, W., Tull, C., Oh, S.-Y., Baldi, P., and Prabhat (2016). Revealing fundamental physics from the Daya Bay Neutrino Experiment using deep neural networks, In <em>2016 15th IEEE International Conference on Machine Learning and Applications</em>, pp.&nbsp;892–897.</li>
<li><strong>Ko, S.</strong> and Won, J.-H. (2016). Processing large-scale data with Apache Spark. <em>Korean Journal of Applied Statistics</em>, 29(6), pp.&nbsp;1077–1094.</li>
<li><strong>Ko, S.</strong>, Jun, G., and Won, J.-H. (2016). HyperConv: Spatio-spectral classification of hyperspectral images with deep convolutional neural networks. <em>Korean Journal of Applied Statistics</em>, 29(5), pp.&nbsp;859–872. [<a href="https://github.com/kose-y/HyperConv">Python code, Theano</a>]</li>
<li>Kim, J., Kim, S., <strong>Ko, S.</strong>, In, Y. H., Moon, H. G., Ahn, S. K., Kim, M. K., Lee, M., Hwang, J. H., Ju, Y. S., Kim, J. I., Noh, D. Y., Kim, S., Park, J. H., Rhee, H., Kim, S., and Han, W. (2015). Recurrent fusion transcripts detected by whole transcriptome sequencing of 120 primary breast cancer samples. <em>Genes, Chromosomes and Cancer</em>, 54(11), pp.&nbsp;681–691.</li>
<li>Kim, Y., Kang, Y. S., Lee, N. Y., Kim, K. Y., Hwang, Y. J., Kim, H. W., Rhyu, I. J., Her, S., Jung, M. K., Kim, S., Lee, C. J., <strong>Ko, S.</strong>, Kowall, N. W., Lee, S. B., Lee, J., and Ryu, H. (2015). Uvrag targeting by Mir125a and Mir351 modulates autophagy associated with Ewsr1 deficiency. <em>Autophagy</em>, 11(5), pp.&nbsp;796–811.</li>
</ul>
</section>
<section id="working-papers-and-preprints" class="level2">
<h2 class="anchored" data-anchor-id="working-papers-and-preprints">Working Papers and Preprints</h2>
<ul>
<li><strong>Ko, S.</strong>, Suchard, M., and Holbrook, A. (2024+). Analyzing millions of SARS-CoV-2 cases with spatiotemporal Hawkes processes.</li>
<li><strong>Ko, S.</strong>, Zhou, H., Sobel, E., and Lange, K. (2024+). Accurate estimation of genetic admixture proportions via haplotype modeling.</li>
<li><strong>Ko, S.</strong>, Zhou, H., Zhou, J., and Won, J.-H. (2024+). DistStat.jl: Towards unified programming for high-performance statistical computing environments in Julia. Under revision. [<a href="https://arxiv.org/abs/2010.16114">Arxiv preprint</a>] [<a href="https://github.com/kose-y/DistStat.jl">Julia code</a>]</li>
</ul>
</section>
<section id="conference-abstracts" class="level2">
<h2 class="anchored" data-anchor-id="conference-abstracts">Conference Abstracts</h2>
<ul>
<li><strong>Ko, S.</strong>, Sobel, E., Zhou, H., and Lange, K., Unsupervised Learning of Ancestry Informative Markers and Genetic Admixture Coefficients From Biobank Data. In <em>Annual Meeting of American Society of Human Genetics 2022</em>, Los Angeles, CA, USA, October 25-29, 2022.</li>
<li><strong>Ko, S.</strong>, Lee, J., Won, J.-H., and Lee, J., Fast and accurate reconstruction for susceptibility source separation in QSM. In Joint Annual Meeting ISMRM-ESMRMB 2018, Paris, France, June 16-21, 2018.</li>
<li>Bhimji, W., Racah, E., <strong>Ko, S.</strong>, Sadowski, P., Tull, C., and Oh, S.-Y., Exploring Raw HEP Data using Deep Neural Networks at NERSC. In <em>38th International Conference on High Energy Physics</em>, Chicago, IL, USA, August 3-10, 2017.</li>
<li><strong>Ko, S.</strong>, Yu, D., and Won, J.-H., A feature-splitting distributed algorithm for generalized linear models under generalized and group lasso penalties. In <em>9th International Conference of the ERCIM Working Group on Computational and Methodological Statistics</em>, Seville, Spain, December 9-11, 2016.</li>
<li>Lee, H. B., Han, W., <strong>Ko, S.</strong>, Kim, M. S., Lim. S., Lee, K. M., Kang, Y. J., Han, J. H., Kim, Y., Yoo, T. K., Moon, H. G., Noh, D. Y., and Kim, S., Identification of ESR splice variants associated with prognosis in estrogen receptor positive breast cancer. In <em>Thirth-Eighth Annual CTRC-AACR San Antonio Breast Cancer Symposium</em>, San Antonio, TX, USA, December 8-12, 2015.</li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
  </div>
  <div class="nav-page nav-page-next">
      <a href="../teaching/index.html" class="pagination-link">
        <span class="nav-page-text">Teaching</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">© Copyright 2024, Seyoon Ko</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">This page is built with <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>



</body></html>