---
title: "Publications"
---

Author profile on Google Scholar: [link](https://scholar.google.com/citations?user=oHe7UR8AAAAJ)

## Peer-reviewed Papers
- Landeros, A., __Ko, S.__, Chang, J.Z., Wu, T.T., Lange, K. (2025). Sparse vertex discriminant analysis: Variable selection for biomedical classification applications. _Computational Statistics & Data Analysis_ 206, 108125. 
- __Ko, S.__, Sobel, E. M., Zhou, H., Lange, K. (2024). Estimation of Genetic Admixture Proportions via Haplotypes. _Computational and Structural Biotechnology Journal_ 23, 4384-4439.
- Chu, B. B., __Ko, S.__, Zhou, J. J., Jensen, A., Zhou, H., Sinsheimer, J. S., Lange, K. (2023). Multivariate Genomewide Association Analysis by Iterative Hard Thresholding. _Bioinformatics_, 39(4), btad193.  [[paper](https://doi.org/10.1093/bioinformatics/btad193)] [[Julia code](https://github.com/OpenMendel/MendelIHT.jl)]
- __Ko, S.__, Chu, B. B., Peterson, D., Okenwa, C., Papp, J. C., Alexander, D. H., Sobel, E. M., Zhou, H., Lange, K. (2023). Unsupervised discovery of ancestry informative markers and genetic admixture proportions in biobank-scale data sets. _American Journal of Human Genetics_, 109(3), pp. 433-445. [[paper](https://doi.org/10.1016/j.ajhg.2022.12.008)][[Julia code](https://github.com/OpenMendel/OpenADMIXTURE.jl)]
- __Ko, S.__, Zhou, H., Zhou, J., and Won, J.-H. (2022). High-Performance Statistical Computing in the Computing Environments of the 2020s. _Statistical Science_, 37(4), pp. 494–518. [[paper](https://projecteuclid.org/journals/statistical-science/volume-37/issue-4/High-Performance-Statistical-Computing-in-the-Computing-Environments-of-the/10.1214/21-STS835.short)] [[Python code, PyTorch, dask](https://github.com/kose-y/dist_stat)]
- Kim J., Jensen, A., __Ko, S.__, Raghavan, S., Phillips, L. S., Hung, A., Sun, Y., Zhou, H., Reaven, P., Zhou, J. J. (2022), Systematic Heritability and Heritability Enrichment Analysis for Diabetes Complications in UK Biobank and ACCORD Studies, _Diabetes_, 71(5), pp. 1137–1148.
- ∗__Ko, S.__, ∗German, C., Jensen, A., Shen, J., Wang, A., Mehrotra, D. V., Sun, Y. V., Sinsheimer, J. S., Zhou, H., Zhou, J. J. (2022), GWAS of longitudinal trajectories at biobank scale, _American Journal of Human Genetics_, 109(3), pp. 433–445. [[Julia code](https://github.com/OpenMendel/TrajGWAS.jl)]
- Chu, B. B., Sobel, E. M., Wasiolek, R., __Ko, S.__, Sinsheimer, J. S., Zhou, H., Lange, K. (2021), A Fast Data-Driven Method for Genotype Imputation, Phasing, and Local Ancestry Inference: MendelImpute.jl. _Bioinformatics_, 37(24), pp. 4756–4763.
- __Ko, S.__, Li, G. X., Choi, H. and Won, J.-H. (2021). Computationally scalable regression modeling for ultrahigh-dimensional omics data with ParProx. _Briefings in Bioinformatics_, 22(6), bbab256. [[Julia code](https://github.com/kose-y/ParProx.jl)]
- Kim, J., Shen, J., Wang, A., Mehrotra, D. V., __Ko, S.__, Zhou, J. J., Zhou, H. (2021), VCSEL: Prioritizing SNP-Set by Penalized Variance Component Selection, _Annals of Applied Statistics_, 15(4), pp. 1652–1672.
- Ryu, E. K., __Ko, S.__, Won, J.-H. (2020). Splitting with Near-Circulant Linear Systems: Applications to Total Variation CT and PET. _SIAM Journal on Scientific Computing_, 42(1), pp. B185–B206. [[Matlab code](https://github.com/kose-y/near-circulant-splitting)]
- Zhou, H., Sinsheimer, J. S., Bates, D. M., Chu, B. B., German, C. A., Ji, S. S., Keys, K. L., Kim, J., __Ko, S.__, Mosher, G. D., and Papp, J. C. (2020). OpenMendel: A cooperative programming project for statistical genetics. _Human Genetics_, 139(1), pp. 61–71. [[project homepage](https://openmendel.github.io/)]
- __Ko, S.__, Yu, D., Won, J.-H. (2019). Easily parallelizable and distributable class of algorithms for structured sparsity, with optimal acceleration. _Journal of Computational and Grahpical Statistics_, 28(4), pp. 821–833. [[Python code, TensorFlow, Docker](https://github.com/kose-y/dist-primal-dual)]
- __Ko, S.__ and Won, J. H. (2019). Optimal Minimization of the Sum of Three Convex Functions with a Linear Operator. In _The 22nd International Conference on Artificial Intelligence and Statistics_, pp. 1185–1194.
- Racah, E., __Ko, S.__, Sadowski, P., Bhimji, W., Tull, C., Oh, S.-Y., Baldi, P., and Prabhat (2016). Revealing fundamental physics from the Daya Bay Neutrino Experiment using deep neural networks, In _2016 15th IEEE International Conference on Machine Learning and Applications_, pp. 892–897.
- __Ko, S.__ and Won, J.-H. (2016). Processing large-scale data with Apache Spark. _Korean Journal of Applied Statistics_, 29(6), pp. 1077–1094.
- __Ko, S.__, Jun, G., and Won, J.-H. (2016). HyperConv: Spatio-spectral classification of hyperspectral images with deep convolutional neural networks. _Korean Journal of Applied Statistics_, 29(5), pp. 859–872. [[Python code, Theano](https://github.com/kose-y/HyperConv)]
- Kim, J., Kim, S., __Ko, S.__, In, Y. H., Moon, H. G., Ahn, S. K., Kim, M. K., Lee, M., Hwang, J. H., Ju, Y. S., Kim, J. I., Noh, D. Y., Kim, S., Park, J. H., Rhee, H., Kim, S., and Han, W. (2015). Recurrent fusion transcripts detected by whole transcriptome sequencing of 120 primary breast cancer samples. _Genes, Chromosomes and Cancer_, 54(11), pp. 681–691.
- Kim, Y., Kang, Y. S., Lee, N. Y., Kim, K. Y., Hwang, Y. J., Kim, H. W., Rhyu, I. J., Her, S., Jung, M. K., Kim, S., Lee, C. J., __Ko, S.__, Kowall, N. W., Lee, S. B., Lee, J., and Ryu, H. (2015). Uvrag targeting by Mir125a and Mir351 modulates autophagy associated with Ewsr1 deficiency. _Autophagy_, 11(5), pp. 796–811.


## Working Papers and Preprints
- __Ko, S.__, Suchard, M., and Holbrook, A. (2024+). Analyzing millions of SARS-CoV-2 cases with
spatiotemporal Hawkes processes.
- __Ko, S.__, Zhou, H., Sobel, E., and Lange, K. (2024+). Accurate estimation of genetic admixture proportions via haplotype modeling.
- __Ko, S.__, Zhou, H., Zhou, J., and Won, J.-H. (2024+). DistStat.jl: Towards unified programming for high-performance statistical computing environments in Julia. Under revision. [[Arxiv preprint](https://arxiv.org/abs/2010.16114)] [[Julia code](https://github.com/kose-y/DistStat.jl)]

## Conference Abstracts
- __Ko, S.__, Sobel, E., Zhou, H., and Lange, K., Unsupervised Learning of Ancestry Informative Markers and Genetic Admixture Coefficients From Biobank Data. In _Annual Meeting of American Society of Human Genetics 2022_, Los Angeles, CA, USA, October 25-29, 2022.
- __Ko, S.__, Lee, J., Won, J.-H., and Lee, J., Fast and accurate reconstruction for susceptibility source separation in QSM. In Joint Annual Meeting ISMRM-ESMRMB 2018, Paris, France, June 16-21, 2018.
- Bhimji, W., Racah, E., __Ko, S.__, Sadowski, P., Tull, C., and Oh, S.-Y., Exploring Raw HEP Data using Deep Neural Networks at NERSC. In _38th International Conference on High Energy Physics_, Chicago, IL, USA, August 3-10, 2017.
- __Ko, S.__, Yu, D., and Won, J.-H., A feature-splitting distributed algorithm for generalized linear models under generalized and group lasso penalties. In _9th International Conference of the ERCIM Working Group on Computational and Methodological Statistics_, Seville, Spain, December 9-11, 2016.
- Lee, H. B., Han, W., __Ko, S.__, Kim, M. S., Lim. S., Lee, K. M., Kang, Y. J., Han, J. H., Kim, Y., Yoo, T. K., Moon, H. G., Noh, D. Y., and Kim, S., Identification of ESR splice variants associated with prognosis in estrogen receptor positive breast cancer. In _Thirth-Eighth Annual CTRC-AACR San Antonio Breast Cancer Symposium_, San Antonio, TX, USA, December 8-12, 2015.
